From 868dde832d4958ff455a1cc501c1a125745f224b Mon Sep 17 00:00:00 2001
From: Eyal Royee <eyalroyee@gmail.com>
Date: Mon, 5 Oct 2020 13:44:47 +0300
Subject: [PATCH] llvm-ld has no soname flag
 llvm-ld don't have the `-soname` flag as gcc's ld, however
 there is a replacement flag called `-install_name`.

Signed-off-by: Eyal Royee <eyalroyee@gmail.com>
---
 dwarf/Makefile | 2 +-
 elf/Makefile   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dwarf/Makefile b/dwarf/Makefile
index 71456bd..f0be64a 100644
--- a/dwarf/Makefile
+++ b/dwarf/Makefile
@@ -33,7 +33,7 @@ to_string.cc: ../elf/enum-print.py dwarf++.hh data.hh Makefile
 CLEAN += to_string.cc
 
 libdwarf++.so.$(SONAME): $(SRCS:.cc=.o)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -Wl,-soname,$@ -o $@ $^
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -Wl,-install_name,$@ -o $@ $^
 CLEAN += libdwarf++.so.*
 
 libdwarf++.so:
diff --git a/elf/Makefile b/elf/Makefile
index 9c7c224..64d628d 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -32,7 +32,7 @@ to_string.cc: enum-print.py data.hh Makefile
 CLEAN += to_string.cc
 
 libelf++.so.$(SONAME): $(SRCS:.cc=.o)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -Wl,-soname,$@ -o $@ $^
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared -Wl,-install_name,$@ -o $@ $^
 CLEAN += libelf++.so.*
 
 libelf++.so:
-- 
2.25.1

